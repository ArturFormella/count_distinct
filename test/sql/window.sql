\set ECHO none
BEGIN;

\i sql/count_distinct--1.3.2.sql

\set ECHO all

select count_distinct(a) over (order by a rows between unbounded preceding and current row)
  from generate_series(1, 20) a;

select count_distinct(a) over (order by a rows between 10 preceding and 10 following row)
  from generate_series(1, 25) a;

ROLLBACK;
