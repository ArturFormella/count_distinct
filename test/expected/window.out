\set ECHO none
select count_distinct(a) over (order by a rows between unbounded preceding and current row)
  from generate_series(1, 20) a;
 count_distinct 
----------------
              1
              2
              3
              4
              5
              6
              7
              8
              9
             10
             11
             12
             13
             14
             15
             16
             17
             18
             19
             20
(20 rows)

select count_distinct(a) over (order by a rows between 10 preceding and 10 following)
  from generate_series(1, 25) a;
 count_distinct 
----------------
             11
             12
             13
             14
             15
             16
             17
             18
             19
             20
             21
             21
             21
             21
             21
             20
             19
             18
             17
             16
             15
             14
             13
             12
             11
(25 rows)

ROLLBACK;
